src/server/scripts/Shadowlands/Sanctum of Domination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,3694 @@
+
+        std::list<Player*> targetList;
+        GetPlayerListInGrid(targetList, GetCaster(), 200.0f);
+
+        for (auto i = targetList.begin(); i != targetList.end(); i++)
+            GetCaster()->CastSpell(*i, SPELL_BANSHEE_WAIL_MISSILE, true);
+    }
+
+    void Register() override
+    {
+        AfterCast += SpellCastFn(spell_sylvanas_windrunner_banshee_wail::HandleAfterCast);
+    }
+};
+
+// Banshee Wail (Marker) - 357719
+class spell_sylvanas_windrunner_banshee_wail_marker : public AuraScript
+{
+    PrepareAuraScript(spell_sylvanas_windrunner_banshee_wail_marker);
+
+    void AfterRemove(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
+    {
+        if (!GetCaster())
+            return;
+
+        GetTarget()->SendPlaySpellVisualKit(VISUAL_KIT_SYLVANAS_BANSHEE_SCREAM_EXPIRE, 0, 0);
+    }
+
+    void Register() override
+    {
+        AfterEffectRemove += AuraEffectRemoveFn(spell_sylvanas_windrunner_banshee_wail_marker::AfterRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// Banshee Wail (Interrupt) - 351252
+class spell_sylvanas_windrunner_banshee_wail_interrupt : public SpellScript
+{
+    PrepareSpellScript(spell_sylvanas_windrunner_banshee_wail_interrupt);
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        if (!GetCaster())
+            return;
+
+        if (GetHitUnit()->HasUnitState(UNIT_STATE_CASTING))
+            GetCaster()->CastSpell(GetHitUnit(), SPELL_BANSHEE_WAIL_SILENCE, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sylvanas_windrunner_banshee_wail_interrupt::HandleHit, EFFECT_0, SPELL_EFFECT_INTERRUPT_CAST);
+    }
+};
+
+// Frigid Shards - 354933
+class spell_sylvanas_windrunner_frigid_shards : public AuraScript
+{
+    PrepareAuraScript(spell_sylvanas_windrunner_frigid_shards);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo ({ SPELL_FRIGID_SHARDS_MISSILE });
+    }
+
+    void HandlePeriodic(AuraEffect const* aurEff)
+    {
+        GetCaster()->CastSpell(GetCaster()->GetVictim(), SPELL_FRIGID_SHARDS_MISSILE, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_sylvanas_windrunner_frigid_shards::HandlePeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// Desecrating Shot - 22400
+struct at_sylvanas_windrunner_disecrating_shot : AreaTriggerAI
+{
+    at_sylvanas_windrunner_disecrating_shot(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnCreate() override
+    {
+        _instance = at->GetInstanceScript();
+
+        _updateDiff = 0;
+    }
+
+    void OnUpdate(uint32 diff) override
+    {
+        if (!_instance)
+            return;
+
+        _updateDiff += diff;
+
+        if (_updateDiff >= 3150)
+        {
+            if (Creature* sylvanas = _instance->GetCreature(DATA_SYLVANAS_WINDRUNNER))
+            {
+                if (!sylvanas)
```

Comment:
you already checking if its null:
`if (Creature* sylvanas = _instance->GetCreature(DATA_SYLVANAS_WINDRUNNER))` implicitly checks if it is null

Reply:
fixed on next commit.